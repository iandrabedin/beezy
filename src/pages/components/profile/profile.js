import React from "react";
import { imageURL, profileSize } from "../../../api";
import "./profile.scss";

export const Profile = props => {
  const { name, character, job, profile_path } = props.profile;

  return (
    <div className="col-sm-12 col-lg-3">
      <div className="profile">
        <div className="profile-image">
          {profile_path !== null ? (
            <img
              src={`${imageURL}${profileSize.big}${profile_path}`}
              alt={name}
            />
          ) : (
            ""
          )}
        </div>
        <div className="profile-title">{name}</div>
        <div className="profile-subtitle">{job ? job : character}</div>
      </div>
    </div>
  );
};
