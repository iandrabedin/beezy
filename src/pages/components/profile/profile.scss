@import "../../../styles/variables.scss";

.profile {
  margin: 0 16px 16px 0;
  text-align: center;

  &-image {
    width: 80px;
    height: 80px;
    margin: auto;
    overflow: hidden;
    border-radius: 50%;
    filter: grayscale(100%);

    img {
      margin-top: -10%;
      width: 100%;
    }
  }

  &-title {
    color: $color-grey-dark;
    font-weight: $text-bold;
    font-size: $font-size-small;
  }

  &-subtitle {
    color: $color-grey;
    font-weight: $text-regular;
    font-size: $font-size-small;
  }
}
