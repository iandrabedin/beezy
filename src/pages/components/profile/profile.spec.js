import React from "react";
import { cleanup, render } from "@testing-library/react";
import { Profile } from "./profile";

afterEach(cleanup);

const props = {
  profile: {
    name: "name",
    job: null,
    character: "character",
    profile_path: "profile_path"
  }
};

describe("Profile", () => {
  it("should display the character", () => {
    const { getByText } = render(<Profile {...props} />);
    expect(getByText("character")).toBeDefined();
  });

  it("should display the job", () => {
    const jobProps = { ...props, profile: { job: "job" } };
    const { getByText } = render(<Profile {...jobProps} />);
    expect(getByText("job")).toBeDefined();
  });

  it("should display the movie poster", () => {
    const { getByAltText } = render(<Profile {...props} />);
    expect(getByAltText("name")).toBeDefined();
  });

  it("should render as expected", () => {
    const { asFragment } = render(<Profile {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
