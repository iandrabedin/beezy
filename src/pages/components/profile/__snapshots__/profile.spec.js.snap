// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render as expected 1`] = `
<DocumentFragment>
  <div
    class="col-sm-12 col-lg-3"
  >
    <div
      class="profile"
    >
      <div
        class="profile-image"
      >
        <img
          alt="name"
          src="https://image.tmdb.org/t/p/w185profile_path"
        />
      </div>
      <div
        class="profile-title"
      >
        name
      </div>
      <div
        class="profile-subtitle"
      >
        character
      </div>
    </div>
  </div>
</DocumentFragment>
`;
